internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -1661,6 +1661,16 @@ func newAssistedServiceDeployment(ctx context.Context, log logrus.FieldLogger, a
 			newSecretEnvVar("POSTGRESQL_DATABASE", "db.name", databaseName),
 			newSecretEnvVar("POSTGRESQL_USER", "db.user", databaseName),
 			newSecretEnvVar("POSTGRESQL_PASSWORD", "db.password", databaseName),
+			// allow auto upgrade when new image is available
+			// allows upgrade for +1 version only, and cannot skip versions
+			corev1.EnvVar{
+				Name:  "POSTGRESQL_UPGRADE",
```

Comment:
What other options we have other than hardlink? Would it change depending on what image we run? Could it be worth having it as env var with default value? So people can technically tweak it (maybe in the next releases of MCE if it's too tight for this one)

Reply:
you have two options
copy: which need 2 PVC and it copies data from PV1 to PV2 as upgraded version
hardlink: migrate the data to the upgrade version 

we agreed on using the `hardlink` way